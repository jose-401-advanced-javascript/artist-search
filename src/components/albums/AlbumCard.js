import React from 'react';
import PropTypes from 'prop-types';
import styles from './AlbumCard.css';
import { Link } from 'react-router-dom';


const AlbumCard = ({ id, title, image, name }) => {

  return (
    <div key={id} className={styles.AlbumCard}>
      <Link to={`/songs/${id}/${name}`}>
        <p>{title}</p>
      </Link>
      <img src={image} />
    </div>
  );
};

AlbumCard.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.string,
  name: PropTypes.string
};

export default AlbumCard;