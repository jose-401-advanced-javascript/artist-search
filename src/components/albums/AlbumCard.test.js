import React from 'react';
import { shallow } from 'enzyme';
import AlbumCard from './AlbumCard';

describe('AlbumCard component', () => {
  it('renders AlbumCard', () => {
    const wrapper = shallow(<AlbumCard id={'asdg'} title={'adshasdh'} image={'adsh'} name={'asdh'}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
