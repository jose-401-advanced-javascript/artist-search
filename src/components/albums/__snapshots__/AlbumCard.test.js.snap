// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumCard component renders AlbumCard 1`] = `
<div
  className="AlbumCard"
  key="asdg"
>
  <Link
    to="/songs/asdg/asdh"
  >
    <p>
      adshasdh
    </p>
  </Link>
  <img
    src="adsh"
  />
</div>
`;
